src/servers/src/postgres/server.rs

```diff
@@ -48,16 +46,17 @@ impl PostgresServer {
         io_runtime: Arc<Runtime>,
         user_provider: Option<UserProviderRef>,
     ) -> PostgresServer {
-        let postgres_handler = Arc::new(PostgresServerHandler::new(query_handler.clone()));
-        let startup_handler = Arc::new(PgAuthStartupHandler::new(
-            user_provider,
-            tls.should_force_tls(),
-            query_handler,
-        ));
+        let make_handler = Arc::new(
+            MakePostgresServerHandlerBuilder::default()
+                .query_handler(query_handler.clone())
+                .user_provider(user_provider.clone())
+                .force_tls(tls.should_force_tls())
+                .build()
+                .unwrap(),
```

Comment:
Is `unwrap` safe here? How do you guarantee that?

Reply:
It's just how `derive_builder` macro designed to use. Also it's relatively safe because on the `unwrap` happens during setup phase and the server will fail fast if there is an error.